### PR TITLE
Update plugin code for use with Mau v3

### PR DESCRIPTION
**Pull request by lgiordani**
This creates a new version of the package that uses Mau v3.

### CODE REVIEW

**Comment by lgiordani** [comment]
@justinmayer when you have 5 minutes this is ready. I had some issues with tests, but I fixed them now. Thanks!

**Comment by lgiordani** [comment]
@justinmayer pretty please? :joy_cat: :pray:

**Review by justinmayer — CHANGES_REQUESTED** [review]
Hey Leonardo. It has been a busier-than-expected summer, so I apologize for the delay in taking a look at this PR.

In addition to the comment I made about not manually fiddling with the version number, there are a number of updated plugin-related "best practices" that I would like to make before the next release of this plugin, assuming that is okay with you. You can see what some of those changes are by looking at the changes I've made to the Cookiecutter Pelican plugin template.

I will make those proposed changes in a separate branch and PR so that you can have a look and tell me what you think.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pelican-mau-reader"
-version = "2.0.0"
+version = "3.0.0"
```
[the remark this answers is not in the document]
Comment (lgiordani) [reply pyproject.toml:3]: @justinmayer thanks, I honestly completely forgot I changed that. I will revert the change. I'll wait for your other PR with the proposed changes, then when that is merged I will rebase this one. Thanks!

[the remark this answers is not in the document]
Comment (justinmayer) [reply pyproject.toml:3]: I submitted #29, which I fear brings changes that will require a non-trivial rebase for you. If that is the case, I'm so sorry! 😅

**Comment by justinmayer** [comment]
The aforementioned PR has been merged, so this PR is ready to be rebased onto current `main` whenever you have a moment. 😊

**Comment by lgiordani** [comment]
@justinmayer I preferred not to rebase, I manually moved my changes to a new branch and force pushed. The amount of changes was limited and I felt it was faster this way. Ready to be reviewed!

**Comment by lgiordani** [comment]
@justinmayer I can't remember the process here. Shall I add here the file to release a major version or shall I first merge and then send a PR with that file?

**Comment by justinmayer** [comment]
While either technique will work equally well, if it were me, I would probably add the `RELEASE.md` file to this branch/PR and then merge it.

**Review by justinmayer — APPROVED** [review]

**Pull request merged**